### ISSUE TITLE
HTTP 500 for update-cloud-config with not interpolated variables

### ISSUE THREAD

**Issue by loewenstein** [cloudfoundry/bosh#1610]
We have a cloud-config containing `dns: [((dns_nameservers))]`

`bosh update-cloud-config <cloud_config>` leads to an HTTP 500 with the following in the director.debug.log:

```
E, [2017-03-06 15:35:32 #15340] [] ERROR -- Director: NetAddr::ValidationError - Could not auto-detect IP version for '((dns_nameservers))'.:
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/netaddr-1.5.1/lib/ip_math.rb:54:in `detect_ip_version'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/netaddr-1.5.1/lib/cidr.rb:157:in `create'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/dns/dns_manager.rb:121:in `block in dns_servers'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/dns/dns_manager.rb:120:in `each'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/dns/dns_manager.rb:120:in `dns_servers'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/deployment_plan/manual_network_subnet.rb:47:in `parse'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/deployment_plan/manual_network.rb:18:in `block in parse'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/deployment_plan/manual_network.rb:17:in `each'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/deployment_plan/manual_network.rb:17:in `parse'
/var/vcap/packages/director/gem_home/ruby/2.3.0/gems/bosh-director-260.0.0/lib/bosh/director/deployment_plan/cloud_manifest_parser.rb:61:in `block in parse_
networks'
```

On the client side:

```
Updating cloud config:
  Director responded with non-successful status code '500' response ''

Exit code 1
```

**Comment by pivotal-jamil-shamy** [comment]
@loewenstein currently we don't support variables interpolation in the Cloud Config. There should be a plan to support it eventually.

**Comment by beyhan** [comment]
We use the bosh-cli for interpolation. Our expectation was to get a better error message. Does it make sense to invest time here for a better error message? Also the return code shouldn't be HTTP 500?

**Comment by pivotal-jamil-shamy** [comment]
@loewenstein @beyhan true, error message could be better. In the meanwhile you can use the new cli to interpolate the cloud-config before uploading it. Example:

`bosh interpolate /path/to/cloud-config.yml -v dns_nameservers="8.8.8.8" >> interpolated-cloud-config.yml`

This command will interpolate the cloud-config, then you can separately upload `interpolated-cloud-config.yml`

Check `bosh interpolate -h` for more options.

**Comment by cppforlife** [comment]
Added story for cloud config interpolation: https://www.pivotaltracker.com/story/show/141743773

**Comment by tjvman** [comment]
Closing - above story has been accepted and is available in `262.0` and up. https://github.com/cloudfoundry/bosh/issues/1637 remains open for tracking bad error handling (unhelpful 500) when uploading malformed cloud configs.
